### PR TITLE
Fix cards responsive issues

### LINKED ISSUE #420: Blog post image previews are distorted on mobile
In the News section. This is most comically noticeable in the 0.8 release notes.
Screenshot_20220730-183202

### PR DESCRIPTION
**Pull request by doup**
Changes:

- Adds a new `link-card` [BEM _component_](https://getbem.com/naming/). This superseed `card` and it's for this image+text combination used in Learn/News/Community.
- Update Learn/News/Community to use `link-card`
- Change Learn to use two columns (following #612)
- Update Learn/Community to load their data from a TOML file
- Extract common "card" styling to a mixin and use it in `asset-card`/`link-card`/`card`.
- Tweak `item-grid` class and use it for 1/2/variable columns. Now it's used almost everywhere.
    - This class controls the spacing between elements, so `margin-bottom` from `card` is not needed anymore (or the `community-card` fix)

For the result, see: https://github.com/bevyengine/bevy-website/pull/422#issuecomment-1455071771

Fixes #420 

## Follow-up tasks

- Remove `card` class and related and instead add: `person-card` & `example-card` following BEM convention.

### CODE REVIEW

**Comment by cart** [comment]
I dig this direction!

**Comment by doup** [comment]
I'm opening this for review. I intended to also migrate `people` and `examples` cards to use BEM… but it's not necessary to do it in this same PR (and would bloat the PR a lot).

**Comment by rparrett** [comment]
The only issue I'm seeing is with news items that don't specify an image in their `extra` metadata

image

File: templates/news.html [hunk 1]
```diff
@@ -2,31 +2,29 @@
...
-  {% endfor %}
+<div class="padded-content item-grid">
+    {% for page in section.pages %}
+        <a class="link-card" href="{{ page.path }}">
+            <div class="link-card__img-wrapper">
+                {% if page.extra.image %}
+                    {% set image_parent = page.path | replace(from="_index.md", to="") %}
+                    <img class="link-card__img" src="{{ image_macros::resize_image(path=image_parent ~ page.extra.image, width=580, height=326) }}" loading="lazy">
+                {% else %}
+                    <img src="/assets/bevy_logo_dark.svg" class="centered-card-image" style="max-width: 15rem;" loading="lazy" />
```
Comment (rparrett) [inline templates/news.html:13]:
```suggestion
                    <img class="link-card__img" src="/assets/bevy_logo_dark.svg" loading="lazy" />
```

Comment (doup) [reply templates/news.html]: Thanks! Good catch 👍

**Review by cart — APPROVED** [review]
Lovely set of changes :heart:

**Pull request merged**